lib/config_files/base_model.rb

```diff
@@ -26,11 +27,41 @@ def save(changes_only: false)
 
     def load
       self.data = @parser.parse(@file_handler.read(@file_path))
+      @loaded = true
+    end
+
+    # powerfull method that sets any value in grub config. It try to be
+    # smart to at first modify existing value, then replace commented out code
+    # and if even that doesn't work, then append it at the end
+    # @note prefer to use specialized methods of children
+    def generic_set(key, value)
+      modify(key, value) || uncomment(key, value) || add_new(key, value)
```

Comment:
what is purpose of having documented default when you use explicit one? It is for me same like having commented out code of parent method in child method which override behavior.